janusgraph-backend-testutils/src/main/java/org/janusgraph/graphdb/JanusGraphIndexTest.java

```diff
@@ -1523,6 +1525,203 @@ public void testIndexParameters() {
         }
     }
 
+    private void checkMixedIndexCountProfiling(TraversalMetrics profile, Map<String, String> annotations) {
+        Metrics metrics = profile.getMetrics(0);
+        assertTrue(metrics.getDuration(TimeUnit.MICROSECONDS) > 0);
+        assertTrue(metrics.getName().contains(JanusGraphMixedIndexCountStep.class.getSimpleName()));
+        Metrics nested = (Metrics) metrics.getNested().toArray()[0];
+        assertEquals(QueryProfiler.MIXED_INEX_COUNT_QUERY, nested.getName());
+        assertTrue(nested.getDuration(TimeUnit.MICROSECONDS) > 0);
+        assertEquals(annotations, nested.getAnnotations());
+    }
+
+    /**
+     * Tests that if a given query can be satisfied by a single mixed index, and it is followed by a count step, then
+     * a count query will be built
+     */
+    @Test
+    public void testMixedIndexQueryFollowedByCount() {
+        final int numV = 100;
+        long total;
+        final String[] strings = {"Uncle Berry has a farm", "and on his farm he has five ducks", "ducks are beautiful animals", "the sky is very blue today"};
+        setupChainGraph(numV, strings, true);
+        clopen();
+
+        /* ===================================================
+        has(key, value) followed by count() can be optimized */
+
+        // vertex
+        total = numV / strings.length * 2;
+        assertEquals(total, graph.indexQuery(VINDEX, "v.text:ducks").vertexTotals());
+        TraversalMetrics profile = graph.traversal().V().has("text", Text.textContains("ducks")).count().profile().next();
+        final String xtextKey = "xtext";
```

Comment:
To pass Solr tests it probably should be:
```
final String xtextKey = getStringField("xtext");
```

Maybe in some other places as well. I didn't check the whole test yet